### ISSUE TITLE
Trajectory Transformer - NameError: name 'self' is not defined

### ISSUE THREAD

**Issue by km5ar** [huggingface/transformers#24455]
### System Info

- `transformers` version: 4.30.2
- Platform: Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.35
- Python version: 3.9.16
- Huggingface_hub version: 0.15.1
- Safetensors version: 0.3.1
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Using GPU in script?:  No
- Using distributed or parallel set-up in script?: No

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

from transformers import TrajectoryTransformerModel
import torch
import numpy as np

device = "cuda"
from transformers import TrajectoryTransformerModel
import torch

model = TrajectoryTransformerModel.from_pretrained(
    "CarlCochet/trajectory-transformer-halfcheetah-medium-v2"
)
model.to(device)
model.eval()

observations_dim, action_dim, batch_size = 17, 6, 256
seq_length = observations_dim + action_dim + 1

trajectories = torch.LongTensor([np.random.permutation(self.seq_length) for _ in range(batch_size)]).to(
    device
)
targets = torch.LongTensor([np.random.permutation(self.seq_length) for _ in range(batch_size)]).to(device)

outputs = model(
    trajectories,
    targets=targets,
    use_cache=True,
    output_attentions=True,
    output_hidden_states=True,
    return_dict=True,
)

### Expected behavior

![1687627979812](https://github.com/huggingface/transformers/assets/54015474/609d3e13-0091-40f3-815f-9cc28d7bc429)

**Comment by sgugger** [comment]
`self` is indeed not defined in your code. Please use the [forums](https://discuss.huggingface.co/) to debug your code :-)

**Comment by km5ar** [comment]
Yeah, but I was using the code directly from the tutorial, so that's the reason I raised this issue...
